### DISCUSSION TITLE
explanation of mmpbsa.in

### DISCUSSION THREAD

**Discussion by drsandhyaks** [Valdes-Tresanco-MS/gmx_MMPBSA#155] [Q&A] [answered]
Dear Valdes,
I am using the gmx_mmpbsa method in gromacs. 
I tested with Spike with antigen from your example given (S1_RBD_with_antibody_6zlr).
It worked well, but I want to know how to conduct binding energy of last 20 ns (Total of 100ns simulation dynamics).
How to define frame for the above? How we know no of frames? ?? From VMD, i have total of 10000 frames. it is correct? Could you explain the input file of mmpbsa.in ?

**Comment by Valdes-Tresanco-MS** [comment, marked as answer]
Please, consult the [documentation](https://valdes-tresanco-ms.github.io/gmx_MMPBSA/getting-started/), in it, you will find the detailed description of the [format, variables, and generation of an `input_file`](https://valdes-tresanco-ms.github.io/gmx_MMPBSA/input_file/). In order to guide you, I will mention a route to follow.
1. Decide which of the [energy calculation methods](https://valdes-tresanco-ms.github.io/gmx_MMPBSA/input_file/#description) is best for you and your system. The spike protein is large so `PB` or `3D-RISM` can take a considerable amount of time. How to [generate the input for the selected method](https://valdes-tresanco-ms.github.io/gmx_MMPBSA/input_file/#generation-of-input-files-with-gmx_mmpbsa)
2. The number of frames is determined from the parameters of your simulation (`nsteps` * `dt`/`nstxout`). Suppose you kept the `dt` at `2 fs`, you would have to have `nsteps = 50,000,000` and `nstxout = 500` to get 10,000 frames in 100ns. Please consult the [GROMACS documentation](https://manual.gromacs.org/), you must not work with these programs as black boxes. Note that **gmx_MMPBSA** is a post-processing program, so it is not responsible for the quality or reliability of the MD
3. The range selection is entirely up to you. For the last 20 ns, the selection should be `startframe=8000`. See [`startframe`, `interval`, and `endframe` variables](https://valdes-tresanco-ms.github.io/gmx_MMPBSA/input_file/#basic-input-options) for more information. Note that [`nmode` has its frames definition](https://valdes-tresanco-ms.github.io/gmx_MMPBSA/input_file/#basic-input-options_3) In the new version, you can select the frame range after the calculation is done using the **gmx_MMPBSA_ana** tool, so you can consider using a more extensive frame range. 
4. Run the calculations. Note that each calculation consumes a different amount of RAM, so make sure you have enough resources for it. Check out [this thread](https://groups.google.com/g/gmx_mmpbsa/c/5Jek4C1DKVo).

We suggest you not use **gmx_MMPBSA** as a black box. We have invested a significant amount of time and resources in the documentation to make it detailed and easy to understand. In it, you will find most of the things you need to use **gmx_MMPBSA**. If you have any specific questions or suggestions, we invite you to open an [issue](https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/issues)

HTH!
Mario S.
